### PR TITLE
[tools] add sync-bundled-native-modules script

### PR DESCRIPTION
**Pull request by dsokal**
# Why

We want to make it easier to update bundledNativeModules.json. 

https://github.com/expo/universe/pull/7426 adds the backend infra for storing the json in the db. This way we can make module upgrades simpler. Expo CLI will use the www endpoint for obtaining the list of native modules compatible with a given SDK version. This will let us suggest users upgrading certain modules. Otherwise, they'd need to upgrade `expo`.

# How

I added the `sync-bundled-native-modules` script to expotools. It allows synchronizing the `bundledNativeModules.json` file with www. It'll be used to inform users about module upgrades without the need to upgrade the `expo` package.

# Test Plan

Screenshot 2021-04-28 at 12

# Checklist

I'll update the release docs in the follow-up PR.

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: tools/src/commands/SyncBundledNativeModules.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      hasChanges = true;
+      logger.info(` - ${npmPackage}: ${chalk.red(`(none)`)} -> ${chalk.green(versionRange)}`);
+    } else if (versionRange !== currentMap[npmPackage].versionRange) {
+      hasChanges = true;
+      logger.info(
+        ` - ${npmPackage}: ${chalk.red(currentMap[npmPackage].versionRange)} -> ${chalk.green(
+          versionRange
+        )}`
+      );
+    }
```
Comment (bbarthec) [inline tools/src/commands/SyncBundledNativeModules.ts:158]:
```suggestion
    // currentMap might either hold a different version of the package or have no entry for given package
    if (versionRange !== currentMap[npmPackage]?.versionRange) {
      hasChanges = true;
      logger.info(` - ${npmPackage}: ${chalk.red(currentMap[npmPackage]?.versionRange ?? `(none)`)} -> ${chalk.green(versionRange)}`);
    }
```
Why to duplicate the very same code block? 🤔

**Review by bbarthec — COMMENTED** [review]
Looking good to me 💪 
Awaiting what other reviewers might say about it 😉

File: tools/src/commands/SyncBundledNativeModules.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+  const sdkVersion = await resolveTargetSdkVersionAsync();
+  const bundledNativeModules = await readBundledNativeModulesAsync();
+  const syncPayload = prepareSyncPayload(bundledNativeModules);
+
+  const currentBundledNativeModules = await getCurrentBundledNativeModules(env, sdkVersion);
+  await compareAndConfirmAsync(currentBundledNativeModules, syncPayload.nativeModules);
+
+  await syncModulesAsync({ env, secret }, sdkVersion, syncPayload);
+  logger.info(chalk.green(`Successfully synced the modules for SDK ${sdkVersion}!`));
```
Comment (tsapeta) [inline tools/src/commands/SyncBundledNativeModules.ts:47]: As the entire log is green, I'd suggest `logger.success`
```suggestion
  logger.success(`Successfully synced the modules for SDK ${sdkVersion}!`);
```

File: tools/src/commands/SyncBundledNativeModules.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    `We need the secret to authenticate you with Expo servers.\nPlease set the ${chalk.green(
+      'EXPO_SDK_NATIVE_MODULES_SECRET'
+    )} env var if you want to skip the prompt in the future.`
+  );
+
+  const { secret } = await inquirer.prompt<{ secret: string }>([
+    {
+      type: 'password',
+      name: 'secret',
+      message: `Secret:`,
```
Comment (tsapeta) [inline tools/src/commands/SyncBundledNativeModules.ts:88]:
```suggestion
      message: 'Secret:',
```

**Review by tsapeta — APPROVED** [review]
Looks good!
Thanks for fixing some linter issues 😅

**Pull request merged**